docs: include template input variables entry in aio glossary (#39861)
angular/angular: aio/content/guide/glossary.md

Read about how to write template expressions in the [template expressions](guide/interpolation#template-expressions) section of the [Interpolation](guide/interpolation) guide.

{@a template-input-variable}

## template input variable

A _template input variable_ is a variable whose value you can reference _within_ a single instance of the template.
You declare a _template input variable_ using the `let` keyword (`let hero`). 

It is used to provide an input in the template of a [structural directive](guide/structural-directives). This would be the template of the HTML tag or component that the structural directive is attached to. This template also defines the boundaries of the scope of the _template input variable_: you may reuse the same variable name again in the definition of other structural directives, as long as they have no overlap with the template of the initial one.

> Note: template _input_ and [_reference_](#template-reference-variable) variable names have separate namespaces. The `hero` in `let hero` is never the same variable as the `hero` declared as `#hero`.

Reviewer: Maybe also add that: _template input variables_ have higher precedence than _template reference variables_

that way it's clear who wins when a conflict occurs...

> Note: template _input_ and [_reference_](#template-reference-variable) variable names have separate namespaces. The `hero` in `let hero` is never the same variable as the `hero` declared as `#hero`.

Reviewer: Maybe also add that: _template input variables_ have higher precedence than _template reference variables_

that way it's clear who wins when a conflict occurs...

> Read more about _template input variable_ in the [Template input variables](guide/built-in-directives#template-input-variables) section of the [Built-in directives](guide/built-in-directives) guide.

Reviewer 2: ```suggestion
For more information, see the [Template input variables](guide/built-in-directives#template-input-variables) section of [Built-in directives](guide/built-in-directives).
```

Read about how to write template expressions in the [template expressions](guide/interpolation#template-expressions) section of the [Interpolation](guide/interpolation) guide.

> {@a template-input-variables}

Reviewer: ```suggestion
{@a template-input-variable}
```

## template input variable

A _template input variable_ is a variable whose value you can reference _within_ a single instance of the template.
It is preceded by the keyword `let` and used to provide an input in the template of a [structural directive](guide/structural-directives) (this would be the template of the HTML tag or component that the structural directive is attached to).

> You declare a _template input variable_ using the `let` keyword (`let hero`).

Reviewer: These two sentences seem to duplicate the same info. Could you please consolidate them into a single sentence?

> You can use the same variable name again in the definition of other structural directives.

Reviewer: Consider redefining this in terms of scope - The boundaries of a template on which a input variable is defined controls the namespace visibility (the variable scope) of this input variable.

Template _input_ and [_reference_](#template-reference-variable) variable names have their own namespaces. The `hero` in `let hero` is never the same

> variable as the `hero` declared as `#hero`.

Reviewer: You won't need this section if you define the scope of the variable as suggested above.
